### PR TITLE
GoL: Add ASCII renderer to Implementing chapter

### PR DESCRIPTION
**Pull request by killercup**
As discussed in [1], introducing the "rendering GoL" while also
introducing raw memory access and rendering using the Canvas API at the
same time might be overwhelming to newcomers.

This commit splits the "Rendering with JavaScript" to first introduce a
simple ASCII renderer (written in Rust so it shows how to get Strings
across to JS), and then in a follow-up "Quicker rendering with
JavaScript" chapter switch to using canvas and ArrayBuffers. We could
split this even further and re-create the ASCII rendering in JS first,
but I leave that for a future PR.

[1]: https://github.com/rust-lang-nursery/rust-wasm/pull/111#discussion_r180518268

### CODE REVIEW

**Comment by killercup** [comment]
If you don't want to split this further, feel free to edit the PR description to add

> Fixes #113

**Comment by est31** [comment]
An alternative that also avoids array communication would probably be to expose a js function that renders one cell on canvas that rust then calls multiple times. But I guess now that you have taken the time to write this all up we should take the strings approach :).

**Comment by killercup** [comment]
@est31 my motivation was to keep the code simple. Providing such a function would not really make this simple enough :)

I also have a local version that is a native binary so you can impl the game and only afterwards add WASM interop. I don't think that is necessary here, though, especially if we can get to a point where people can interactively play round with this.

File: src/game-of-life/implementing.md [hunk 1]
```diff
@@ -270,10 +268,37 @@ impl Universe {
...
+[Display]: https://doc.rust-lang.org/1.25.0/std/fmt/trait.Display.html
+
+```rust
+use std::fmt;
+
+impl fmt::Display for Universe {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        for line in self.cells.as_slice().chunks(self.width as usize) {
+            for &cell in line {
+                write!(f, "{}", if cell == Cell::Dead { "◻️" } else { "◼️" })?;
```
[the remark this answers is not in the document]
Comment (killercup) [reply src/game-of-life/implementing.md]: rustfmt writes it like

```rust
                write!(
                    f,
                    "{}",
                    if cell == Cell::Dead {
                        "◻️"
                    } else {
                        "◼️"
                    }
                )?;
```

but I really don't want that. How about this?

```rust
let symbol = if cell == Cell::Dead { "◻️" } else { "◼️" };
write!(f, "{}", symbol)?;
```

File: src/game-of-life/implementing.md [hunk 2]
```diff
@@ -353,7 +369,87 @@ than the old `greet` function:
...
+const pre = document.getElementById("game-of-life-canvas");
+const universe = Universe.new();
+```
+
+The JavaScript runs in a `requestAnimationFrame` loop. On each iteration, it
+draws the current universe to the `<pre>`, and then calls `Universe::tick`.
+
+```js
+const renderLoop = () => {
+  pre.innerText = universe.render();
```
[the remark this answers is not in the document]
Comment (killercup) [reply src/game-of-life/implementing.md]: Wow, I don't think I've ever used this, and don't even think it was available when I first did stuff that should've used it… (And now you know how long I've been doing this JS stuff 😄)

File: src/game-of-life/implementing.md [hunk 3]
```diff
@@ -270,10 +268,37 @@ impl Universe {
...
+        Ok(())
+    }
+}
+```
+
 Finally, we define a constructor that initializes the universe with an
-interesting pattern of live and dead cells, as well as getters for a universe's
-width, height, and pointer to its cells array. All of these are exposed to
-JavaScript as well.
+interesting pattern of live and dead cells, as well as getters
```
[the remark this answers is not in the document]
Comment (killercup) [reply src/game-of-life/implementing.md]: Oh, probably a copypasta thing. Changed it to

> Finally, we define a constructor that initializes the universe with an
> interesting pattern of live and dead cells, as well as a `render` method:

**Pull request merged**